### DISCUSSION TITLE
user from inventory file is not working

### DISCUSSION THREAD

**Discussion by bessarabov** [umputun/spot#21] [General]
Here is an example that shows my problem.

```
$ cat spot.yml
user: asdf
ssh_key: ~/.ssh/id_rsa

targets:

tasks:
  one:
    commands:
      - script: ls /tmp
$ cat inventory
192.168.31.86 pi
$
```

I'm trying to use this inventory file (the idea is to run all the tasks on all the hosts that are specified in `inventory`), but I get the error:

```
$ spot -f spot.yml --inventory-file=inventory
simplotask v0.5.0-992f291-2023-04-30T22:56:15Z
failed: can't run task "one" for target "default": 1 error(s) occurred: [0] {can't connect to 192.168.31.86:22: failed to create client connection to 192.168.31.86:22: failed to create client connection to 192.168.31.86:22: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain}
```

It looks like the username from `inventory` is not used by the spot. When I explicitly pass user to the spot everyting works well (this is the same user that I  specified in `inventory` file)

```
$ spot -f spot.yml --inventory-file=inventory --user=pi
simplotask v0.5.0-992f291-2023-04-30T22:56:15Z
[192.168.31.86:22] run task one, commands: 1
[192.168.31.86:22] completed  {script: sh -c "ls /tmp"} (1.005s)
[192.168.31.86:22] completed task one, commands: 1
[INFO] completed: hosts:1, commands:1 in 1.1s
```

**Comment by umputun** [comment]
thx for the report. I'm in the middle of refactoring all this format for inventory to make it yml. I will check after the change what's going on with the user.

**Comment by bessarabov** [comment]
Thank you, in 0.7.2 with new yaml inventory this is working as expected. I'm closing this converstation.
